Cannot generate PDFs with Sphinx version 1.6.1 on Windows. Perl Missing.

Subject: Cannot generate PDFs with Sphinx version 1.6.1 on Windows. Perl Missing.

### Problem
- After upgrading to Sphinx 1.6.1 from 1.5.x, I am no longer able to generate pdf documents with `make latexpdf` which was working before. The problem is that `latexmk` is now in use and requires `perl` to be installed. I don't have `perl` installed and I am unable to install additional software as this is a work computer.

I have looked around, Google, and here on the issues list, and I'm unable to find anything that works. The command `make latexpdf PDFLATEX=pdflatex` as found [here](https://github.com/sphinx-doc/sphinx/issues/2513#issuecomment-216229066), doesn't work - it gives exactly the same errors about `perl` being missing. (See below)

#### Procedure to reproduce the problem
```
make latexpdf
```

#### Error logs / results
```
Running Sphinx v1.6.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
processing UpdatingSuperBASICManual.tex...index Part1 p1_intro p1_software Part2 p2_download Part3 p3_rst Part4 P4_intro

resolving references...
writing... done
copying images... [100%] images\LogIssue.png
copying TeX support files...
done
build succeeded.
latexmk -pdf -dvi- -ps-  'UpdatingSuperBASICManual.tex'
latexmk: The script engine could not be found.
latexmk: Data: scriptEngine="perl.exe"
Makefile:33: recipe for target 'UpdatingSuperBASICManual.pdf' failed
mingw32-make: *** [UpdatingSuperBASICManual.pdf] Error 1

Build finished; the PDF files are in build/latex.
```
#### Expected results
I would expect that when I request a pdf to be built, it would be built. As it did in the previous version, or, in the event that a component of the build has changed dramatiacally, as in this case, that the install/upgrade notes would make the additional dependencies known, and/or an option to use a different utility be supplied to revert to the previous working operation.

Windows isn't the best of operating systems for having decent utilities installed by default! ;-)

### Reproducible project / your project
- Not required? Any project will do, just don't have `perl` installed. 

### Environment info
- OS: Windows 7. 64 bit.
- Python version: 3.5.1
- Sphinx version: 1.6.1
- MikTex: 2.9

### Workaround?
I have a workaround which I didn't need to use under Sphinx 1.5.x:

````
make latexpdf
...
cd build\latex
pdflatex whatever.tex
````
That generates the pdf as desired but using `pdflatex` as provided by MikTex.  I have seen a mention _somewhere_ that MikTex provides `perl` but at least at version 2.9, it does not.

@jfbu I tried `make latexpdf` with sphinx/doc on Windows.

(Sphinx stable HEAD (and 1.6.1) still have #3817 issue, however I bypassed it temporary) to check latexpdf target.)

> I doubt it is 100% Windows compatible especially for the case of Japanese documents because I was told that syntax such as ... in latexmkjarc would not work on Windows. 

Indeed. It failed.

```
make[1]: ディレクトリ `C:/Project/sphinx-dev/sphinx/doc/_build/latex' に入ります
extractbb 'agogo.png'
extractbb 'alabaster.png'
extractbb 'bizstyle.png'
extractbb 'classic.png'
extractbb 'haiku.png'
extractbb 'more.png'
extractbb 'nature.png'
extractbb 'pyramid.png'
extractbb 'scrolls.png'
extractbb 'sphinx.png'
extractbb 'sphinx_rtd_theme.png'
extractbb 'sphinxdoc.png'
extractbb 'traditional.png'
extractbb 'translation.png'
for f in *.pdf; do extractbb "$f"; done
f の使い方が誤っています。
make[1]: *** [sphinx.pdf] エラー 255
make[1]: ディレクトリ `C:/Project/sphinx-dev/sphinx/doc/_build/latex' から出ます```
```
It means
```
for f in *.pdf; do extractbb "$f"; done
The syntax of the command is incorrect
make[1]: *** [sphinx.pdf] Error 255
```

--- a/CHANGES
+++ b/CHANGES
@@ -35,7 +35,7 @@ Bugs fixed
 * #3803: Sphinx crashes with invalid PO files
 * #3791: PDF "continued on next page" for long tables isn't internationalized
 * #3788: smartquotes emits warnings for unsupported languages
-* #3807: docs do not say Latexmk is dependency on Windows for ``make latexpdf``
+* #3807: latex Makefile for ``make latexpdf`` is only for unixen
 * #3781: double hyphens in option directive are compiled as endashes
 * #3817: latex builder raises AttributeError
 
@@ -54,7 +54,7 @@ Dependencies
 * (updated) latex output is tested with Ubuntu trusty's texlive packages (Feb.
   2014) and earlier tex installations may not be fully compliant, particularly
   regarding Unicode engines xelatex and lualatex
-* (added) latexmk is required for ``make latexpdf``
+* (added) latexmk is required for ``make latexpdf`` on GNU/Linux and Mac OS X
   (refs: #3082)
 
 Incompatible changes

--- a/doc/builders.rst
+++ b/doc/builders.rst
@@ -200,11 +200,12 @@ The builder's "name" must be given to the **-b** command-line option of
          Formerly, testing had been done for some years on Ubuntu precise
          (based on TeXLive 2009).
       .. versionchanged:: 1.6
-         Use of ``latexmk`` for ``make latexpdf``
+         Use of ``latexmk`` for ``make latexpdf`` on GNU/Linux and Mac OS X
 
       Since 1.6, ``make latexpdf`` (or
       ``make -C "<builddir>/latex"`` after a ``sphinx-build`` run) uses
-      ``latexmk``. One can pass to ``latexmk`` options via the ``LATEXMKOPTS``
+      ``latexmk`` (not on Windows).
+      One can pass to ``latexmk`` options via the ``LATEXMKOPTS``
       Makefile variable. For example:
 
       .. code-block:: console

--- a/sphinx/builders/latex.py
+++ b/sphinx/builders/latex.py
@@ -218,6 +218,12 @@ class LaTeXBuilder(Builder):
                 copy_asset_file(path.join(staticdirname, filename),
                                 self.outdir, context=context)
 
+        # use pre-1.6.x Makefile for make latexpdf on Windows
+        if os.name == 'nt':
+            staticdirname = path.join(package_dir, 'texinputs_win')
+            copy_asset_file(path.join(staticdirname, 'Makefile_t'),
+                            self.outdir, context=context)
+
         # copy additional files
         if self.config.latex_additional_files:
             logger.info(bold('copying additional files...'), nonl=1)

--- a/sphinx/texinputs_win/Makefile_t
+++ b/sphinx/texinputs_win/Makefile_t
@@ -1,0 +1,86 @@
+# Makefile for Sphinx LaTeX output
+
+ALLDOCS = $(basename $(wildcard *.tex))
+ALLPDF = $(addsuffix .pdf,$(ALLDOCS))
+ALLDVI = $(addsuffix .dvi,$(ALLDOCS))
+ALLPS  = $(addsuffix .ps,$(ALLDOCS))
+
+# Prefix for archive names
+ARCHIVEPREFIX =
+# Additional LaTeX options
+LATEXOPTS =
+# format: pdf or dvi
+FMT = pdf
+
+LATEX = latex
+PDFLATEX = {{ latex_engine }}
+MAKEINDEX = makeindex
+
+{% if latex_engine == 'platex' %}
+all: all-pdf-ja
+all-pdf: all-pdf-ja
+{% else %}
+all: $(ALLPDF)
+all-pdf: $(ALLPDF)
+{% endif -%}
+all-dvi: $(ALLDVI)
+all-ps: $(ALLPS)
+
+all-pdf-ja:
+	for f in *.pdf *.png *.gif *.jpg *.jpeg; do extractbb $$f; done
+	for f in *.tex; do platex -kanji=utf8 $(LATEXOPTS) $$f; done
+	for f in *.tex; do platex -kanji=utf8 $(LATEXOPTS) $$f; done
+	for f in *.tex; do platex -kanji=utf8 $(LATEXOPTS) $$f; done
+	-for f in *.idx; do mendex -U -f -d "`basename $$f .idx`.dic" -s python.ist $$f; done
+	for f in *.tex; do platex -kanji=utf8 $(LATEXOPTS) $$f; done
+	for f in *.tex; do platex -kanji=utf8 $(LATEXOPTS) $$f; done
+	for f in *.dvi; do dvipdfmx $$f; done
+
+zip: all-$(FMT)
+	mkdir $(ARCHIVEPREFIX)docs-$(FMT)
+	cp $(ALLPDF) $(ARCHIVEPREFIX)docs-$(FMT)
+	zip -q -r -9 $(ARCHIVEPREFIX)docs-$(FMT).zip $(ARCHIVEPREFIX)docs-$(FMT)
+	rm -r $(ARCHIVEPREFIX)docs-$(FMT)
+
+tar: all-$(FMT)
+	mkdir $(ARCHIVEPREFIX)docs-$(FMT)
+	cp $(ALLPDF) $(ARCHIVEPREFIX)docs-$(FMT)
+	tar cf $(ARCHIVEPREFIX)docs-$(FMT).tar $(ARCHIVEPREFIX)docs-$(FMT)
+	rm -r $(ARCHIVEPREFIX)docs-$(FMT)
+
+gz: tar
+	gzip -9 < $(ARCHIVEPREFIX)docs-$(FMT).tar > $(ARCHIVEPREFIX)docs-$(FMT).tar.gz
+
+bz2: tar
+	bzip2 -9 -k $(ARCHIVEPREFIX)docs-$(FMT).tar
+
+xz: tar
+	xz -9 -k $(ARCHIVEPREFIX)docs-$(FMT).tar
+
+# The number of LaTeX runs is quite conservative, but I don't expect it
+# to get run often, so the little extra time won't hurt.
+%.dvi: %.tex
+	$(LATEX) $(LATEXOPTS) '$<'
+	$(LATEX) $(LATEXOPTS) '$<'
+	$(LATEX) $(LATEXOPTS) '$<'
+	-$(MAKEINDEX) -s python.ist '$(basename $<).idx'
+	$(LATEX) $(LATEXOPTS) '$<'
+	$(LATEX) $(LATEXOPTS) '$<'
+
+%.pdf: %.tex
+	$(PDFLATEX) $(LATEXOPTS) '$<'
+	$(PDFLATEX) $(LATEXOPTS) '$<'
+	$(PDFLATEX) $(LATEXOPTS) '$<'
+	-$(MAKEINDEX) -s python.ist '$(basename $<).idx'
+	$(PDFLATEX) $(LATEXOPTS) '$<'
+	$(PDFLATEX) $(LATEXOPTS) '$<'
+
+%.ps: %.dvi
+	dvips '$<'
+
+clean:
+	rm -f *.log *.ind *.aux *.toc *.syn *.idx *.out *.ilg *.pla *.ps *.tar *.tar.gz *.tar.bz2 *.tar.xz $(ALLPDF) $(ALLDVI)
+
+.PHONY: all all-pdf all-dvi all-ps clean zip tar gz bz2 xz
+.PHONY: all-pdf-ja
+

